### ISSUE TITLE
Block Kit support

### ISSUE THREAD

**Issue by seratch** [slackapi/slack-api-specs#18]
### Description

The documentation already supports Block Kit. However, the changes to parameters/response are not yet reflected in this repository. Do you have any plans to work on it in the short term? If I can do something for it, I am keen to be involved in it.

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [ ] bug
- [ ] enhancement (feature request)
- [ ] question
- [x] documentation related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/slack-api-specs/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

**Comment by episod** [comment]
We want to offer full JSON schema (and a subset within our OpenAPI specs) for Block Kit but aren't yet ready to do so. In the meantime, the newest version of the spec on master includes a bare bones schema for `blocks` and includes the `blocks` parameter on methods that support them. Look for deeper support (including the difference between input and output blocks) in the future.

**Comment by simonh1000** [comment]
it would be wonderful if you could open source the validation code in https://app.slack.com/block-kit-builder

**Comment by x3ro** [comment]
Curious to see that there hasn't been any notice on this for almost four years. As far as I can see, Block Kit is still the way to go for displaying UI in Slack, right? Having no way to validate it makes for fairly poor developer experience 😕

**Comment by riffraff** [comment]
it's quite hard to plan an integration and be sure it will work with real data without a schema, it would be great if you could publish it.

**Comment by fab-mindflow** [comment]
+1
It makes a lot of sense to add BlockKit JSON schema to OpenAPI specifications. Any news on this?

**Comment by fab-mindflow** [comment]
Btw, it looks like this JSON schema exists:
https://gist.github.com/renatorib/1fb1a9bd71435b41bee602d15bc56899

Would it be possible to share this JSON schema officially?

**Comment by delanni** [comment]
The schema that's mentioned in the gist, and here by @fab-mindflow is incomplete, it cuts out after ~18k characters, so it's an invalid schema right now.

**Comment by fab-mindflow** [comment]
This was first open in 2019. Can someone at Slack take a look please?

**Comment by lawrencegripper** [comment]
Nudging this as it's super painful having to manually copy paste to the block builder website or only find out in a live system. I'd like to wrap my block generation client side in a unit test which validates the output against the JSON schema. 

Given the tooling on the block kit builder has this to validate against I was hoping it's easy to publish :pray:

**Comment by lawrencegripper** [comment]
I'm not someone at slack but took a look!

TLDR: With some manual effort and the typescript type information I semi-auto generated the schema. [Here it is for others to use.](https://gist.github.com/lawrencegripper/ee24242249fc6a1a0a73c561766dd486) Note: It validates an array of blocks for input.

```json
 [
    {
	    "type": "header",
	    "text": {
		    "type": "plain_text",
		    "text": "some message",
		    "emoji": true
	    }
     }
]
```

**How?:**

Talking with @itoys he suggested [this approach using the typescript definitions](https://blog.dennisokeeffe.com/blog/2021-12-09-generating-json-schema-from-typescript-types) for blocks to create a schema.

Doing that over this file gave me the definitions for the block types available :partying_face: 
- https://github.com/slackapi/node-slack-sdk/blob/main/packages/types/src/block-kit/blocks.ts

I then needed a top level definition for the to match the possible different block types, I did this with `if-then` syntax for json schema.

```json
    "type": "object",
    "properties": {
      "type": {
        "enum": [
          "image",
          "context",
          "actions",
          "divider",
          "section",
          "input",
          "file",
          "header",
          "video",
          "rich_text"
        ]
      }
    },
    "required": [
      "type"
    ],
    "additionalProperties": true,
    "allOf": [
      {
        "if": {
          "properties": {
            "type": {
              "const": "image"
            }
          }
        },
        "then": {
          "$ref": "#/definitions/ImageBlock"
        }
      },
      {
        "if": {
          "properties": {
            "type": {
              "const": "context"
            }
          }
        },
        "then": {
          "$ref": "#/definitions/ContextBlock"
        }
      },
```

**Comment by lawrencegripper** [comment]
Thanks to @okeeffed for the typescript to json schema code and awesome blog 🙇‍♂️

**Comment by okeeffed** [comment]
Stoked that it managed to help out @lawrencegripper ❤️

**Comment by davidcelis** [comment]
That schema is unfortunately also incomplete; many of Slack's blocks have limits on how many elements can appear in a list (e.g. an upper limit of 100 options in a dropdown select's `options` field) or how long various strings can be (e.g. a 150 character limit for the text element in a header block) or even how many blocks a single definition is limited to.
